### ISSUE TITLE
[clang-tidy] bugprone-lambda-function-name false positive

### ISSUE THREAD

**Issue by 1aam2am1** [llvm/llvm-project#89065]
Hi, I'm fixing my code and using this check for wrong function name prints, but fixing this error creates false positive.

How to reproduce:
```cpp
#include <iostream>
// Type your code here, or load an example.
int bug(int num) {
    return [num](){
        std::cout << __func__ << "\n";
        return num;
    }();
}

int fixed(int num) {
    return [num, func=__func__](){
        std::cout << func << "\n";
        return num;
    }();
}

int main()
{
    bug(1);
    fixed(1);
    return 0;
}
```
Execute clang-tydy, with: 

 -checks=-*,bugprone-lambda-function-name

Result:

```
<[source>:5:22: warning: inside a lambda, '__func__' expands to the name of the function call operator; consider capturing the name of the enclosing function explicitly [bugprone-lambda-function-name]]
    5 |         std::cout << __func__ << "\n";
      |                      ^
[<source>:11:23: warning: inside a lambda, '__func__' expands to the name of the function call operator; consider capturing the name of the enclosing function explicitly [bugprone-lambda-function-name]]
   11 |     return [num, func=__func__](){
      |
```

Code execution result:
```
operator()
fixed
```

We can see that 'bug' function have a error and it is found.
But we can see that the fix in 'fixed' function, where we are capturing __func__ name inside lambda capture is also flagged, and it should not be.
